app/views/articles/_search.html.erb

```diff
@@ -1,199 +1,199 @@
-<script defer>
```

Comment:
Great catch on these `script` tags with `defer` and no `src` attribute!

I can see we have some other usages in the codebase with `<script async>` and no `src` attribute which will have the same problem - I'll make a small issue for these because we'd definitely want to correct this.

FYI @nickytonline - what do you think about the approach here? The alternative would be to bring the script content out into its own js file and reference with the `src` attribute.

Reply:
agree with you, I wanted to do this part on another PR cause I'm not sure of the final behaviour.

Reply:
in the meantime, I replaced all  `defer` attributes on inline scripts with `DOMContentLoaded` events @aitchiss @nickytonline

Reply:
I think this breaks the search: 

1. start this branch
2. got to the homepage
3. type something in the search box

the page transitions to http://localhost:3000/search?q=QUERY but nothing happens. I think it's because we have InstantClick which does not trigger the `DOMContentLoad`. See http://instantclick.io/scripts

If you then reload the page, you'll see the results.

If I replace 

```js
window.addEventListener('DOMContentLoaded', function() {
```

with 

```js
InstantClick.on('change', function() {
```

the scenario works, but then it breaks on page load, again, because there's no page load event. 

See also https://docs.forem.com/frontend/instant-click/

I think we should migrate this to its own file or disable instant click.

I reckon this applies to most some of the other `<script defer>` changes as well.

Reply:
to be fixed in another PR/issue